### PR TITLE
C# TOC and #if article improvements

### PR DESCRIPTION
**Pull request by gewarren**
- add display names to TOC for filter use (e.g. I typed in "preprocessor" and nothing came up)
- improve formatting/readbility of one sentence in #if topic

### CODE REVIEW

File: docs/csharp/toc.yml [hunk 1]
```diff
@@ -93,6 +95,7 @@
 - name: What's new in C#
   items:
   - name: C# 8.0
+    displayName: what's new
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]: Why lower-case?

```suggestion
    displayName: What's new
```

File: docs/csharp/toc.yml [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
   items:
   - name: Overview
+    displayName: get started
     href: getting-started/index.md
   - name: Introduction to the C# language and the .NET framework
     href: getting-started/introduction-to-the-csharp-language-and-the-net-framework.md
 - name: Tutorials
   items:
   - name: Overview
+    displayName: tutorials
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
    displayName: Tutorials
```

File: docs/csharp/toc.yml [hunk 3]
```diff
@@ -317,23 +322,26 @@
       href: programming-guide/inside-a-program/identifier-names.md
     - name: C# Coding Conventions
       href: programming-guide/inside-a-program/coding-conventions.md
-  - name: Main() and Command-Line Arguments
+  - name: Main() and command-line arguments
     items:
     - name: Overview
+      displayName: main method
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
      displayName: Main method
```

File: docs/csharp/toc.yml [hunk 4]
```diff
@@ -317,23 +322,26 @@
...
+  - name: Programming concepts
     items:
     - name: Overview
+      displayName: programming concepts
       href: programming-guide/concepts/index.md
-    - name: Asynchronous Programming with async and await
+    - name: Asynchronous programming with async and await
       items:
       - name: Overview
+        displayName: asynchronous programming
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
        displayName: Asynchronous programming
```

File: docs/csharp/toc.yml [hunk 5]
```diff
@@ -366,6 +374,7 @@
     - name: Attributes
       items:
       - name: Overview
+        displayName: attributes
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
        displayName: Attributes
```

File: docs/csharp/toc.yml [hunk 6]
```diff
@@ -377,31 +386,33 @@
...
-    - name: Caller Information
+    - name: Caller information
       href: programming-guide/concepts/caller-information.md
     - name: Collections
       href: programming-guide/concepts/collections.md
-    - name: Covariance and Contravariance
+    - name: Covariance and contravariance
       items:
       - name: Overview
+        displayName: covariance and contravariance
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
        displayName: Covariance and contravariance
```

File: docs/csharp/toc.yml [hunk 7]
```diff
@@ -377,31 +386,33 @@
...
+        - name: Use Variance in Delegates
           href: programming-guide/concepts/covariance-contravariance/using-variance-in-delegates.md
-        - name: Using Variance for Func and Action Generic Delegates
+        - name: Use Variance for Func and Action Generic Delegates
           href: programming-guide/concepts/covariance-contravariance/using-variance-for-func-and-action-generic-delegates.md
-    - name: Expression Trees
+    - name: Expression trees
       items:
       - name: Overview
+        displayName: expression trees
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
        displayName: Expression trees
```

File: docs/csharp/toc.yml [hunk 8]
```diff
@@ -892,16 +904,18 @@
     - name: Serialization (C#)
       items:
       - name: Overview
+        displayName: serialization
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
        displayName: Serialization
```

File: docs/csharp/toc.yml [hunk 9]
```diff
@@ -892,16 +904,18 @@
...
         href: programming-guide/concepts/serialization/how-to-write-object-data-to-an-xml-file.md
       - name: "How to read object data from an XML file"
         href: programming-guide/concepts/serialization/how-to-read-object-data-from-an-xml-file.md
       - name: "Walkthrough: Persisting an Object in Visual Studio"
         href: programming-guide/concepts/serialization/walkthrough-persisting-an-object-in-visual-studio.md
-  - name: Statements, Expressions, and Operators
+  - name: Statements, expressions, and operators
     items:
     - name: Overview
+      displayName: statements expressions operators
```
Comment (Youssef1313) [inline docs/csharp/toc.yml]:
```suggestion
      displayName: Statements, expressions, and operators
```

**Comment by Youssef1313** [comment]
If you think my suggestions are correct, there are more occurrences of the same kind of the suggestions above.

File: docs/csharp/toc.yml (hunk 9 above)
Comment (BillWagner) [reply docs/csharp/toc.yml]: Here, I think the commas may be necessary for the drop down to find each one separately. But I'm not sure. We needed them when we added multi-character operators.

File: docs/csharp/toc.yml [hunk 10]
```diff
@@ -317,23 +322,26 @@
...
       href: programming-guide/main-and-command-args/command-line-arguments.md
     - name: "How to display command-line arguments"
       href: programming-guide/main-and-command-args/how-to-display-command-line-arguments.md
     - name: Main() Return Values
       href: programming-guide/main-and-command-args/main-return-values.md
-  - name: Programming Concepts
+  - name: Programming concepts
     items:
     - name: Overview
+      displayName: programming concepts
```
Comment (BillWagner) [inline docs/csharp/toc.yml]: This is a really smart idea whenever we've added a node for the new TOC structure. Should we do the same in other areas? /cc @mairaw

**Review by BillWagner — APPROVED** [review]
Overall, this looks great @gewarren 

You can :shipit: once we address the remaining questions.

File: docs/csharp/toc.yml (hunk 2 above)
Comment (gewarren) [reply docs/csharp/toc.yml]: No, the capitalization doesn't matter. It's not actually displayed anywhere (despite the name).

File: docs/csharp/toc.yml (hunk 10 above)
Comment (gewarren) [reply docs/csharp/toc.yml]: Yes, it's too bad the TOC filter doesn't pick up keywords in titles that don't have a doc linked to them.

File: docs/csharp/toc.yml (hunk 1 above)
Comment (gewarren) [reply docs/csharp/toc.yml]: Despite the name, this text isn't displayed anywhere, so casing is irrelevant.

File: docs/csharp/toc.yml (hunk 2 above)
Comment (Youssef1313) [reply docs/csharp/toc.yml]: @gewarren, The name gave me the impression that it's displayed somewhere. I'm curious what this key does ?

Comment (gewarren) [reply docs/csharp/toc.yml]: It's basically to add keywords to the node. The TOC filter shows that node if the user filters on that keyword.

**Pull request merged**